mysys/mf_keycache.c

```diff
@@ -2857,17 +2853,21 @@ uchar *simple_key_cache_read(SIMPLE_KEY_CACHE_CB *keycache,
       }
       if (!(block->status & BLOCK_ERROR))
       {
-        if (page_st != PAGE_READ)
+        if (page_st == PAGE_TO_BE_READ)
+        {
+          MYSQL_KEYCACHE_READ_MISS();
+          read_block_primary(keycache, block,
+                     keycache->key_cache_block_size, read_length+offset);
+        }
+        else if (page_st != PAGE_READ)
```

Comment:
I'd rather have ```page_st == PAGE_WAIT_TO_BE_READ``` here, but this form should work as well.

Reply:
Yeh, I was trying avoid changing too many branches